src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1805,7 +1805,11 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
             $transportDefinition = (new Definition(TransportInterface::class))
                 ->setFactory([new Reference('messenger.transport_factory'), 'createTransport'])
                 ->setArguments([$transport['dsn'], $transport['options'] + ['transport_name' => $name], new Reference($serializerId)])
-                ->addTag('messenger.receiver', ['alias' => $name])
+                ->addTag('messenger.receiver', [
+                        'alias' => $name,
+                        'failure_transport' => $transport['failure_transport'] ?? null,
+                    ]
+                )
```

Comment:
I think we can simplify (the complication is based off of my original, not perfect understanding), which will *really* simplify things in MessengerPass.

On a high level, what we care about is this: "can this receiver also be used as a failure receiver?". And we care about this for one simple (not-that-important) reason: so that we know if it can be used on the `messenger:failed:*` commands with the new `--failure-transport=` option.

So, instead of setting `failure_transport => 'some string'`, which communicates which failure transport this receiver is assigned to (I think we don't actually care about that), what if the new attribute is just `failure_transport => true/false` (and it would default to false if not set).

Basically, we would determine if this receiver is being used as a failure transport or not. If it is, we add `failure_transport = true`. Then, in `MessengerPass`, we can read this option. We would use the `alias` to know the *name* of this failure transport.

To do this, we'll need to do some more work here in `FrameworkExtension`. I think we may need move the 2nd `foreach ($config['transports'] as $name => $transport) {` that you have below to be ABOVE this section, so that we can create a map/array of which transports have at least one other transport using them as a `failure_transport` (taking into account, of course, the global failure transport). Then, inside this loop, we can use that map/array to set the `failure_transport = true/false` alias option.

THEN, way over in `MessengerPass`, we would create the `$failureTransportsMap` a bit more like how we currently create the `$receiverMapping` variable inside the `foreach ($container->findTaggedServiceIds($this->receiverTag) as $id => $tags) {` - something like this:

```diff
$receiverMapping[$id] = new Reference($id);
foreach ($tags as $tag) {
    if (isset($tag['alias'])) {
        $receiverMapping[$tag['alias']] = $receiverMapping[$id];

+         // I'm requiring the "alias" to be set for failure_transport to work
+         // that's probably ok... but we could also be smarter and use
+         // $tag['alias'] OR fallback to $id, like is done with $receiverMapping
+         if (isset($tag['failure_transport']) && null !== $tag['failure_transport']) {
+             $failureTransportsMap[$tag['alias']] = $receiverMapping[$id];
+         }
    }
}
```